### PR TITLE
New Memset module to manage DNS zone domains

### PR DESCRIPTION
**Pull request by glitchcrab**
##### SUMMARY
This PR provides a module to manage DNS zone domains hosted by [Memset](https://www.memset.com/apidocs/methods_dns.html#dns.zone_domain_create) .

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
 * memset_zone_domain

##### ANSIBLE VERSION
```
2.6
```

### CODE REVIEW

**Comment by duncan9562** [comment]
Successfully created a new Zone Domain in Memset's DNS manager.

**Comment by ghost** [comment]
works fine for me.

**Comment by PebbleOfParagon** [comment]
Successfully created/removed Zone Domain in DNS Manager

File: lib/ansible/modules/cloud/memset/memset_zone_domain.py [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+            retvals[val] = eval(val)
+
+    return(retvals)
+
+
+def main():
+    global module
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(required=False, default='present', choices=['present', 'absent'], type='str'),
```
Comment (sac) [inline lib/ansible/modules/cloud/memset/memset_zone_domain.py]: required=False is not needed. You can remove that.
`state=dict(type='str', choices=['absent', 'present'], default='present'),`  should be good enough.

File: lib/ansible/modules/cloud/memset/memset_zone_domain.py [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+
+def api_validation(args=None):
+    '''
+    Perform some validation which will be enforced by Memset's API (see:
+    https://www.memset.com/apidocs/methods_dns.html#dns.zone_domain_create)
+    '''
+    # zone domain length must be less than 250 chars
+    if len(args['domain']) > 250:
+        stderr = 'Zone domain must be less than 250 characters in length.'
+        module.fail_json(failed=True, msg=stderr, stderr=stderr)
```
Comment (sac) [inline lib/ansible/modules/cloud/memset/memset_zone_domain.py]: Just curious, why do you want the error message twice? Anyway msg would be printed.

**Review by sac — COMMENTED** [review]
Other than a couple of nits LGTM (Haven't tested the module)

File: lib/ansible/modules/cloud/memset/memset_zone_domain.py (hunk 2 above)
Comment (glitchcrab) [reply lib/ansible/modules/cloud/memset/memset_zone_domain.py]: Good point, that was on my snagging list to resolve. I've just pushed a change to fix this and another occurrence.

File: lib/ansible/modules/cloud/memset/memset_zone_domain.py (hunk 1 above)
Comment (glitchcrab) [reply lib/ansible/modules/cloud/memset/memset_zone_domain.py]: Just pushed a commit to remove required.

**Review by sac — APPROVED** [review]

**Comment by maxamillion** [comment]
@resmo you assigned this for self-review but haven't updated since the initial comment, anything you'd like to see addressed?

**Review by resmo — APPROVED** [review]
LGTM

**Comment by James-MS** [comment]
Works for me!

**Pull request merged**